### PR TITLE
🐛 Fix display of default value for Enum parameters inside of a list, include docs and tests

### PR DESCRIPTION
**Pull request by asieira**
Ensures the enum `.value` is used in the generated help text for default Argument and Option values. Supports both isolated enum instances and lists or tuples of enum instances.

Did local testing with existing scripts/pytest.sh and all passed.

Fixes https://github.com/tiangolo/typer/issues/290

### CODE REVIEW

**Comment by jacksonwb** [comment]
Any movement here? Also experiencing this.

**Comment by svlandeg** [comment]
Sorry for the delay on this! We're going through the PR backlog and hope to look into this one soonish. Thanks for the contribution and your patience!

**Review by svlandeg — COMMENTED** [review]
Thanks again for the contribution! 

I ran some local tests and this looks good to me. The PR needs some tests though - I'll go ahead and add those by pushing to your branch directly, hope that's OK!

File: tests/test_tutorial/test_parameter_types/test_enum/test_tutorial001.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ def test_help():
     assert result.exit_code == 0
     assert "--network" in result.output
     assert "[simple|conv|lstm]" in result.output
+    assert "default: simple" in result.output
```
Comment (svlandeg) [inline tests/test_tutorial/test_parameter_types/test_enum/test_tutorial001.py:20]: Note that the docs in fact already document this behaviour (first bash script on [enum page](https://typer.tiangolo.com/tutorial/parameter-types/enum/)):
>   --network [simple|conv|lstm]  [default: simple]

But on `master` you currently get
> --network        [simple|conv|lstm]  [default: NeuralNetwork.simple]  

So the docs don't require further updating - this PR really just ensures the correct behaviour as it was already documented.

File: tests/test_tutorial/test_parameter_types/test_enum/test_tutorial003.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+app = typer.Typer()
+app.command()(mod.main)
+
+
+def test_help():
+    result = runner.invoke(app, ["--help"])
+    assert result.exit_code == 0
+    assert "--groceries" in result.output
+    assert "[Eggs|Bacon|Cheese]" in result.output
+    assert "default: Eggs, Cheese" in result.output
```
Comment (svlandeg) [inline tests/test_tutorial/test_parameter_types/test_enum/test_tutorial003.py:20]: on `master`, this would be:

> default: Food.food_1, Food.food_3

**Comment by tiangolo** [comment]
Awesome, thanks @asieira! :rocket: 

And thanks a lot @svlandeg for the help! :bow: 

This will be available in the next hours in Typer 0.9.2 :rocket:

**Pull request merged**